src/kv/untyped_map_handle.h

```diff
@@ -260,5 +266,53 @@ namespace kv::untyped
 
       return size_;
     }
+
+#ifdef KV_STATE_RB
```

Comment:
I think this doesn't actually need to be gated by `KV_STATE_RB`, since it should also work with the champ map as currently written? The only change for the RB is that we can early out when we get past the range (which we can't do for the champ, since seeing one item past the range doesn't mean we won't see future entries within the range). This would be neater, as we can then use the API elsewhere without gating on this seemingly unrelated `KV_STATE_RB` define. Of course as soon as we start implementing the optimisations listed below the implementations will become even more divergent, but while we can keep the API stable it seems worth doing?